File: IMDSSample.pl
#!/usr/bin/env perl

use strict;
use warnings;

use HTTP::Tiny;
use JSON;
use Data::Dumper;

my $md_url = 'http://169.254.169.254/metadata/instance?api-version=2021-02-01';
my $headers = { Metadata => 'true' };

# Proxies must be bypassed when calling Azure IMDS
my $response = HTTP::Tiny->new(proxy => undef)->get($md_url, { headers => $headers });
my $info = decode_json($response->{ content });

print Dumper($info);

print Dumper($info->{ compute }->{ vmId });

File: IMDSSample.vb
Imports System.IO
Imports System.Net
Imports Newtonsoft.Json

Module Module1

    Sub Main()
        Dim request As HttpWebRequest = WebRequest.Create("http://169.254.169.254/metadata/instance?api-version=2019-03-11")
        ' IMDS requires proxies to be bypassed
        Dim proxy As New WebProxy()
        request.Headers.Add("Metadata: True")
        request.Proxy = proxy
        Dim response As HttpWebResponse = request.GetResponse()
        Dim dataStream As Stream = response.GetResponseStream()
        Dim reader As New StreamReader(dataStream)
        Dim responseFromServer As String = reader.ReadToEnd()

        Dim json As String = JsonConvert.SerializeObject(JsonConvert.DeserializeObject(responseFromServer), Formatting.Indented)
        Console.WriteLine(json)
        reader.Close()
        dataStream.Close()
        response.Close()
    End Sub

End Module